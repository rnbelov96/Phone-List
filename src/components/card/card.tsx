import React from 'react';
import { connect } from 'react-redux';
import { AppActionCreators } from '../../reducer/app/app';
import { DataActionCreators } from '../../reducer/data/data';
import { PhotoType } from '../../types/general-types';

export type CardProps = {
  photoData: PhotoType;

  setModalWindowUrl: (url: string | null) => void;
  deletePhoto: (id: number) => void;
};

export const PureCard: React.FunctionComponent<CardProps> = ({
  photoData,
  setModalWindowUrl,
  deletePhoto,
}: CardProps) => {
  const [isImageLoading, setImageLoadingStatus] = React.useState<boolean>(true);

  return (
    <div className="card list__item">
      <img
        onClick={() => {
          setModalWindowUrl(photoData.url);
          if (window.innerWidth > document.body.clientWidth) {
            document.body.style.paddingRight = `${window.innerWidth - document.body.clientWidth}px`;
          }
          document.body.style.overflowY = 'hidden';
        }}
        src={photoData.thumbnailUrl}
        className="card-img-top list__item-img"
        alt="small square"
        style={{
          minHeight: '150px',
          display: isImageLoading ? 'none' : 'block',
        }}
        onLoad={() => setImageLoadingStatus(false)}
      />
      <div
        style={{
          minHeight: '150px',
          display: isImageLoading ? 'block' : 'none',
        }}
        className="lds-ring mx-auto list__item-circle"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <div className="card-body list__item-content">
        <h5 className="card-title list__item-title">{photoData.title}</h5>
        <button
          onClick={() => {
            deletePhoto(photoData.id);
          }}
          type="button"
          className="btn btn-danger list__item-btn"
        >
          Delete Photo
        </button>
      </div>
    </div>
  );
};

export default connect(() => ({}), {
  setModalWindowUrl: AppActionCreators.setModalWindowUrl,
  deletePhoto: DataActionCreators.deletePhoto,
})(PureCard);
