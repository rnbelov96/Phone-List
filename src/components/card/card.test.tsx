import React from 'react';
import renderer from 'react-test-renderer';
import { PureCard } from './card';

test('Card should render correctly', () => {
  const tree = renderer
    .create(
      <PureCard
        photoData={{
          albumId: 1,
          id: 1,
          thumbnailUrl: 'some url',
          title: 'Some title',
          url: 'some url',
        }}
        deletePhoto={() => {}}
        setModalWindowUrl={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
