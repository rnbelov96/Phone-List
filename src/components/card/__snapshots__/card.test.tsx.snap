// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="card list__item"
>
  <img
    alt="small square"
    className="card-img-top list__item-img"
    onClick={[Function]}
    onLoad={[Function]}
    src="some url"
    style={
      Object {
        "display": "none",
        "minHeight": "150px",
      }
    }
  />
  <div
    className="lds-ring mx-auto list__item-circle"
    style={
      Object {
        "display": "block",
        "minHeight": "150px",
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>
  <div
    className="card-body list__item-content"
  >
    <h5
      className="card-title list__item-title"
    >
      Some title
    </h5>
    <button
      className="btn btn-danger list__item-btn"
      onClick={[Function]}
      type="button"
    >
      Delete Photo
    </button>
  </div>
</div>
`;
